lib/auth.js

```diff
@@ -0,0 +1,453 @@
+'use strict';
+const _ = require('lodash'),
+  passport = require('passport'),
+  users = require('./services/users'),
+  db = require('./services/db'),
+  references = require('./services/references'),
+  session = require('express-session'),
+  flash = require('express-flash'),
+  handlebars = require('handlebars'),
+  basicAuth = require('basic-auth'),
+  fs = require('fs'),
+  path = require('path');
+
+/**
+ * determine if a route is protected
+ * protected routes are ?edit=true and any method other than GET
+ * @param {object} req
+ * @returns {boolean}
+ */
+function isProtectedRoute(req) {
+  return !!req.query.edit || req.method !== 'GET';
```

Comment:
I'm not a huge fan of truthy values in url params but if we're already treating it truthy its good for me